src/main/scala/Parser.scala

```diff
@@ -121,6 +121,10 @@ private class FuseParser extends RegexParsers with PackratParsers {
     "bool" ^^ { _ => TBool() } |
     "bit" ~> angular(number) ^^ { case s => TSizedInt(s, false) } |
     "ubit" ~> angular(number) ^^ { case s => TSizedInt(s, true) } |
+    "fix" ~> angular(number~","~number ) ^^{ case s1~_~s2 => TFixed(s1,s2,false) } |
+    "fix" ~> angular(number) ^^{ case s => TFixed(s, s, false) } | 
```

Comment:
What’s the difference between these two? Is the second one the same as a bit.

If so, either unify bit and fix or require two arguments.

Reply:
That's the default of ap_fixed. There exists both ap_int type and ap_fixed type. If you define ap_fixed<3>, it is basically the same as ap_fixed<3,3> in HLS, but it's allowed. Just like you can say double x = 1.0, which is just the same value as int x = 1. But the type is different.

Reply:
Can we just have the first format (I.e. both Params are required). I don’t like the inconsistency of two interfaces for the fixed type. For polymorphism, it would make more sense to have one representation.

Reply:
In fact, fixed point can require more arguments (for overflow and quantization mode), but I just haven't implemented. I don't feel any necessity to remove it just like you can define double x = 1.0.

Reply:
> I don't feel any necessity to remove it just like you can define double x = 1.0.

I don’t follow this.

There is a difference between the interface exposed by C++ and a Dahlia type. When you define a type T<M>, you’re describing a type level functions T that takes one parameter (M) and returns a new type. In C++, templates describe compile time computation that is more flexible than a normal type system which lets you write the two forms.

Having this means you have two function fix with different arities. Dahlia tries to expose an explicit interface to all the C++ HLS stuff which is why I want one of the two. You can represent the second form with the first one so why have two forms?